homeassistant/components/shelly/__init__.py

```diff
@@ -158,7 +158,11 @@ def _async_input_events_handler(self):
             last_event_count = self._last_input_events_count.get(channel)
             self._last_input_events_count[channel] = block.inputEventCnt
 
-            if last_event_count == block.inputEventCnt or event_type == "":
+            if (
+                not last_event_count
```

Comment:
Let's be more explicit checking against `None` so we don't accidentally filter out 0

```suggestion
                last_event_count is None
```